### PR TITLE
make sure bottom nav bar always contributes labels

### LINKED ISSUE #19900: BottomNavigationBar: unselected tab should have tab name as a11y label
Open the Bottom navigation demo in the gallery. Use talkback to focus on an unselected tab. It should announce the name of that tab, but it doesn't.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/19900

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]

**Comment by gspencergoog** [comment]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -866,7 +881,7 @@ class RenderAnimatedOpacity extends RenderProxyBox {
 
   @override
   void visitChildrenForSemantics(RenderObjectVisitor visitor) {
-    if (child != null && _alpha != 0)
+    if (alwaysIncludeSemantics || (child != null && _alpha != 0))
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: Fixed

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 2]
```diff
@@ -792,7 +792,12 @@ class RenderAnimatedOpacity extends RenderProxyBox {
   /// Creates a partially transparent render object.
   ///
   /// The [opacity] argument must not be null.
-  RenderAnimatedOpacity({ @required Animation<double> opacity, RenderBox child }) : super(child) {
+  RenderAnimatedOpacity({
+    @required Animation<double> opacity,
+    @required bool alwaysIncludeSemantics,
+    RenderBox child,
+  }) : _alwaysIncludeSemantics = alwaysIncludeSemantics,
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: Fixed

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 3]
```diff
@@ -792,7 +792,12 @@ class RenderAnimatedOpacity extends RenderProxyBox {
   /// Creates a partially transparent render object.
   ///
   /// The [opacity] argument must not be null.
-  RenderAnimatedOpacity({ @required Animation<double> opacity, RenderBox child }) : super(child) {
+  RenderAnimatedOpacity({
+    @required Animation<double> opacity,
+    @required bool alwaysIncludeSemantics,
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter/lib/src/rendering/proxy_box.dart]: Fixed

**Review by amirh — APPROVED** [review]

**Pull request merged**